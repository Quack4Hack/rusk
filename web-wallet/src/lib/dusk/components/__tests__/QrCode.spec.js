import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
import { cleanup, render } from "@testing-library/svelte";

import { QrCode } from "..";

describe("QrCode", () => {
  vi.useFakeTimers();

  const baseProps = {
    value: "some text",
  };
  const baseOptions = {
    props: baseProps,
    target: document.body,
  };
  const toDataUrlSpy = vi.spyOn(HTMLCanvasElement.prototype, "toDataURL");

  afterEach(() => {
    cleanup();
    toDataUrlSpy.mockClear();
  });

  afterAll(() => {
    toDataUrlSpy.mockRestore();
    vi.useRealTimers();
  });

  it("should render the QrCode component and update it when any of the prop change", async () => {
    const { container, rerender } = render(QrCode, baseOptions);

    await vi.runAllTimersAsync();

    expect(container.firstChild).toMatchSnapshot();
    expect(toDataUrlSpy).toHaveBeenCalledTimes(1);

    await rerender({ value: "some different text" });

    expect(toDataUrlSpy).toHaveBeenCalledTimes(2);

    await rerender({ bgColor: "#000" });

    expect(toDataUrlSpy).toHaveBeenCalledTimes(3);

    await rerender({ qrColor: "#fff" });

    expect(toDataUrlSpy).toHaveBeenCalledTimes(4);

    await rerender({ width: 500 });

    expect(toDataUrlSpy).toHaveBeenCalledTimes(5);
  });
});
