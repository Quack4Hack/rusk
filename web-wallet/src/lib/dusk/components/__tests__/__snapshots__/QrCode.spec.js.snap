// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QrCode > should render the QrCode component 1`] = ``;
